c++/src/capnp/compat/json.c++

```diff
@@ -804,11 +814,7 @@ private:
 
     KJ_REQUIRE(numArrayPtr.size() > 0, "Expected number in JSON input.");
 
-    kj::Vector<char> number;
-    number.addAll(numArrayPtr);
-    number.add('\0');
-
-    return kj::String(number.releaseAsArray());
+    return kj::heapString(numArrayPtr.begin(), numArrayPtr.size());
```

Comment:
I wonder why this function is returning a string at all. Why not parse the number to `double` here, to avoid the need for a heap allocation?

Reply:
`parseDouble` takes a StringPtr, so this will require either an allocation or adding an API for ArrayPtr-based parsing. I did run a quick test and doing the latter improves the benchmark result by 1.5%, I added a TODO for now.

Reply:
Could use `strtod()` directly to parse from an array.

Reply:
The current approach ends up calling `NoLocaleStrtod` in string.cpp, which provides locale handling, I assumed that's also needed for JSON.